macros/src/main/scala/gestalt/macros/DefMacros.scala

```diff
@@ -20,6 +20,26 @@ object plusObject {
         toolbox.Lit(null)
     }
   }
+
+  inline def varargs(tped: Int*): Int = meta {
+    val q"$items: $_" = tped
+    items match {
+      case toolbox.SeqLiteral(items: Seq[toolbox.Tree]) =>
+        items.reduceLeft((a, b) => q"$a + $b")
+      case _ =>
+        q"$items.reduce((a:Int,b:Int)=> a + b)"
+    }
+  }
```

Comment:
Handling at least two cases separately is inevitable. It is just the nature of varargs. The most accurate type of a vararg is `Either[Tree,Seq[Tree]]`. It is also possible to use `Pair[Tree,Seq[Tree]]` and just put `Nil` when it is not a sequence literal. N.B.: It is possible to have multiple sets of varargs if the macro is defined as a curried function.